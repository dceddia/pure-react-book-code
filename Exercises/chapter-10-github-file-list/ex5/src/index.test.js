import ReactDOM from 'react-dom';
jest.mock('react-dom', () => ({ render: jest.fn() }));
global.document.querySelector = () => {};

import { StoryList, stories } from './index';
import React from 'react';
import renderer from 'react-test-renderer';

it('renders', () => {
  const tree = renderer.create(<StoryList stories={stories} />).toJSON();
  expect(tree).toMatchSnapshot();
});
