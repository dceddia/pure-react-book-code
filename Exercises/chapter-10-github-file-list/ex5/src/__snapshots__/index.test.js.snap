// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders 1`] = `
<ol
  className="stories"
  start={1}
>
  <li>
    <span
      className="story"
    >
      <h3
        className="title"
      >
        Salesforce buys word processing app Quip for $750M
      </h3>
      <div
        className="subtext"
      >
        <span
          className="score"
        >
          154
           points
        </span>
         by 
        <span
          className="user"
        >
          rhc2104
        </span>
         
        <span
          className="time"
        >
          2 years ago
        </span>
         | 
        <a>
          flag
        </a>
         | 
        <a>
          hide
        </a>
         | 
        <a>
          84
           comments
        </a>
      </div>
    </span>
  </li>
  <li>
    <span
      className="story"
    >
      <h3
        className="title"
      >
        YouTube's road to HTTPS
      </h3>
      <div
        className="subtext"
      >
        <span
          className="score"
        >
          180
           points
        </span>
         by 
        <span
          className="user"
        >
          seanwatson
        </span>
         
        <span
          className="time"
        >
          2 years ago
        </span>
         | 
        <a>
          flag
        </a>
         | 
        <a>
          hide
        </a>
         | 
        <a>
          63
           comments
        </a>
      </div>
    </span>
  </li>
  <li>
    <span
      className="story"
    >
      <h3
        className="title"
      >
        Seymour Papert has died
      </h3>
      <div
        className="subtext"
      >
        <span
          className="score"
        >
          487
           points
        </span>
         by 
        <span
          className="user"
        >
          j4mie
        </span>
         
        <span
          className="time"
        >
          2 years ago
        </span>
         | 
        <a>
          flag
        </a>
         | 
        <a>
          hide
        </a>
         | 
        <a>
          89
           comments
        </a>
      </div>
    </span>
  </li>
  <li>
    <span
      className="story"
    >
      <h3
        className="title"
      >
        FCC Reaches Settlement with TP-Link in Wi-Fi Router Investigation
      </h3>
      <div
        className="subtext"
      >
        <span
          className="score"
        >
          83
           points
        </span>
         by 
        <span
          className="user"
        >
          pentago
        </span>
         
        <span
          className="time"
        >
          2 years ago
        </span>
         | 
        <a>
          flag
        </a>
         | 
        <a>
          hide
        </a>
         | 
        <a>
          23
           comments
        </a>
      </div>
    </span>
  </li>
  <li>
    <span
      className="story"
    >
      <h3
        className="title"
      >
        Ask HN: Who is hiring? (August 2016)
      </h3>
      <div
        className="subtext"
      >
        <span
          className="score"
        >
          368
           points
        </span>
         by 
        <span
          className="user"
        >
          whoishiring
        </span>
         
        <span
          className="time"
        >
          2 years ago
        </span>
         | 
        <a>
          flag
        </a>
         | 
        <a>
          hide
        </a>
         | 
        <a>
          620
           comments
        </a>
      </div>
    </span>
  </li>
  <li>
    <span
      className="story"
    >
      <h3
        className="title"
      >
        The vanishing civility: The game of jerks, bozos and assholes
      </h3>
      <div
        className="subtext"
      >
        <span
          className="score"
        >
          95
           points
        </span>
         by 
        <span
          className="user"
        >
          azuajef
        </span>
         
        <span
          className="time"
        >
          2 years ago
        </span>
         | 
        <a>
          flag
        </a>
         | 
        <a>
          hide
        </a>
         | 
        <a>
          73
           comments
        </a>
      </div>
    </span>
  </li>
  <li>
    <span
      className="story"
    >
      <h3
        className="title"
      >
        Surprises of the Faraday Cage
      </h3>
      <div
        className="subtext"
      >
        <span
          className="score"
        >
          273
           points
        </span>
         by 
        <span
          className="user"
        >
          leephillips
        </span>
         
        <span
          className="time"
        >
          2 years ago
        </span>
         | 
        <a>
          flag
        </a>
         | 
        <a>
          hide
        </a>
         | 
        <a>
          46
           comments
        </a>
      </div>
    </span>
  </li>
  <li>
    <span
      className="story"
    >
      <h3
        className="title"
      >
        Sentiment Analysis – Improving Bayesian Methods
      </h3>
      <div
        className="subtext"
      >
        <span
          className="score"
        >
          27
           points
        </span>
         by 
        <span
          className="user"
        >
          KennyCason
        </span>
         
        <span
          className="time"
        >
          2 years ago
        </span>
         | 
        <a>
          flag
        </a>
         | 
        <a>
          hide
        </a>
         | 
        <a>
          12
           comments
        </a>
      </div>
    </span>
  </li>
  <li>
    <span
      className="story"
    >
      <h3
        className="title"
      >
        Netflix Billing Migration to AWS – Part III
      </h3>
      <div
        className="subtext"
      >
        <span
          className="score"
        >
          63
           points
        </span>
         by 
        <span
          className="user"
        >
          samber
        </span>
         
        <span
          className="time"
        >
          2 years ago
        </span>
         | 
        <a>
          flag
        </a>
         | 
        <a>
          hide
        </a>
         | 
        <a>
          29
           comments
        </a>
      </div>
    </span>
  </li>
  <li>
    <span
      className="story"
    >
      <h3
        className="title"
      >
        Reverse engineering and removing Pokémon GO’s certificate pinning
      </h3>
      <div
        className="subtext"
      >
        <span
          className="score"
        >
          92
           points
        </span>
         by 
        <span
          className="user"
        >
          mkane848
        </span>
         
        <span
          className="time"
        >
          2 years ago
        </span>
         | 
        <a>
          flag
        </a>
         | 
        <a>
          hide
        </a>
         | 
        <a>
          33
           comments
        </a>
      </div>
    </span>
  </li>
</ol>
`;
